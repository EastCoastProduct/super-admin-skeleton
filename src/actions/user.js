import { USER_FETCH_SUCCESS, USER_GET_FAILED } from '../constants/actions';
import { API_URL } from '../constants/application';
import createFormData from '../utils/createFormData';
import parseErrors from '../utils/parseErrors';
import fetch from '../utils/fetch';

export const userFetchSuccess = profile => ({
  type: USER_FETCH_SUCCESS,
  profile,
});

export const userGetFailed = error => ({
  type: USER_GET_FAILED,
  error,
});

export const userGetFetch = userId =>
  dispatch =>
    fetch(`${API_URL}/users/${userId}`).then(resp =>
      dispatch(userFetchSuccess(resp))
    ).catch(err =>
      dispatch(userGetFailed(err.message))
    );

export const userUpdateFetch = (values, userId, cb) =>
  dispatch =>
    fetch(`${API_URL}/users/${userId}`, {
      method: 'POST',
      body: createFormData(values),
    }).then((resp) => {
      dispatch(userFetchSuccess(resp));
      return typeof cb === 'function' && cb();
    }).catch(err =>
      Promise.reject(parseErrors(err))
    );

export const userCreateFetch = (values, cb) =>
  dispatch =>
    fetch(`${API_URL}/superAdmin/users`, {
      method: 'POST',
      body: JSON.stringify(values),
    }).then((resp) => {
      dispatch(userFetchSuccess(resp));
      return typeof cb === 'function' && cb(resp.id);
    }).catch(err =>
      Promise.reject(parseErrors(err))
    );
