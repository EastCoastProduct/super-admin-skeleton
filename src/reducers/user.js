import { fromJS } from 'immutable';
import { USER_FETCH_SUCCESS, USER_GET_FAILED, USER_CHANGE_SUCCESS }
  from '../constants/actions';

const initialState = fromJS({
  error: null,
  profile: {},
});

export default (state = initialState, action) => {
  switch (action.type) {
    case USER_FETCH_SUCCESS:
      return state.merge(fromJS({
        error: null,
        profile: action.profile,
      }));
    case USER_CHANGE_SUCCESS:
      return state.merge(fromJS({
        error: null,
        profile: state.get('profile').merge(...action.responses),
      }));
    case USER_GET_FAILED:
      return state.merge(fromJS({
        error: action.error,
      }));
    default:
      return state;
  }
};
