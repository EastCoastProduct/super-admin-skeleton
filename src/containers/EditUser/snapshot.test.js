import React from 'react';
import renderer from 'react-test-renderer';
import { fromJS } from 'immutable';
import { StyleSheetTestUtils } from 'aphrodite/no-important';
import { fullProfile, profile } from '../../fixtures/user';
import { EditUserComponent } from './';

jest.mock('redux-form/immutable', () => ({
  Field: 'Field',
  reduxForm: () => component => component,
}));

describe('EditUser component snapshot', () => {
  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });
  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('renders required data', () => {
    const tree = renderer.create(
      <EditUserComponent
        dispatch={() => {}}
        form="EditUser"
        handleSubmit={f => f => f}
        params={{ userId: 1 }}
        profile={fromJS(profile)}
        router={{}}
        submitting={true}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders required data with error', () => {
    const tree = renderer.create(
      <EditUserComponent
        dispatch={() => {}}
        error="User not found."
        form="EditUser"
        handleSubmit={f => f => f}
        params={{ userId: 999 }}
        profile={fromJS(fullProfile)}
        router={{}}
        submitting={false}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
