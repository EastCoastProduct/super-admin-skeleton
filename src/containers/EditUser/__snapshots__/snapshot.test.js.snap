exports[`EditUser component snapshot renders required data 1`] = `
<main>
  <h1
    className="header_fffb07">
    <span
      className="headerText_104b05c">
      Users
    </span>
     > 
    Edit Profile
  </h1>
  <div
    className="box_5m47zw">
    <form
      noValidate={true}
      onSubmit={[Function]}>
      <Field
        component={[Function]}
        id="EditUser"
        image={undefined}
        name="image"
        validated={true} />
      <Field
        component={[Function]}
        id="EditUser"
        label="First Name"
        maxLength="30"
        name="firstname"
        placeholder="First Name"
        type="text" />
      <Field
        component={[Function]}
        id="EditUser"
        label="Last Name"
        maxLength="30"
        name="lastname"
        placeholder="Last Name"
        type="text" />
      <Field
        component={[Function]}
        id="EditUser"
        label="Bio"
        maxLength="1000"
        name="bio"
        placeholder="Bio"
        textarea={true} />
      <div>
        <button
          className="button_w1w7v0-o_O-button_1q6mnjs"
          disabled={true}
          type="submit">
          Save
        </button>
        <a
          className="link_1mjr7jy"
          onClick={[Function]}
          style={Object {}}>
          Cancel
        </a>
      </div>
    </form>
  </div>
</main>
`;

exports[`EditUser component snapshot renders required data with error 1`] = `
<main>
  <h1
    className="header_fffb07">
    <span
      className="headerText_104b05c">
      Users
    </span>
     > 
    John Doe
  </h1>
  <div
    className="box_5m47zw">
    <form
      noValidate={true}
      onSubmit={[Function]}>
      <Field
        component={[Function]}
        id="EditUser"
        image="https://s3.amazonaws.com/ecp-boilerplate/images/9049ffe0-b70f-11e6-92c0-51aa519864b8.jpg"
        name="image"
        validated={true} />
      <Field
        component={[Function]}
        id="EditUser"
        label="First Name"
        maxLength="30"
        name="firstname"
        placeholder="First Name"
        type="text" />
      <Field
        component={[Function]}
        id="EditUser"
        label="Last Name"
        maxLength="30"
        name="lastname"
        placeholder="Last Name"
        type="text" />
      <Field
        component={[Function]}
        id="EditUser"
        label="Bio"
        maxLength="1000"
        name="bio"
        placeholder="Bio"
        textarea={true} />
      <div>
        <button
          className="button_w1w7v0-o_O-button_1q6mnjs"
          disabled={false}
          type="submit">
          Save
        </button>
        <a
          className="link_1mjr7jy"
          onClick={[Function]}
          style={Object {}}>
          Cancel
        </a>
      </div>
      <p
        className="error_8cisgn">
        <i
          className="icon_1w16ibx fa fa-exclamation-triangle" />
        User not found.
      </p>
    </form>
  </div>
</main>
`;
